app/services/badges/award_contributor_from_github.rb

```diff
@@ -9,19 +15,51 @@ class AwardContributorFromGithub
       "forem/DEV-ios",
     ].freeze
 
-    def self.call(since = 1.day.ago, msg = I18n.t("services.badges.thank_you"))
-      return unless (badge_id = Badge.id_for_slug(BADGE_SLUG))
+    def self.call(msg = I18n.t("services.badges.thank_you"))
+      new(msg).call
+    end
+
+    def initialize(msg)
+      @msg = msg
+      @badge_slugs_with_id = get_badge_slugs_with_id
+    end
 
+    def call
       REPOSITORIES.each do |repo|
-        commits = Github::OauthClient.new.commits(repo, since: since.utc.iso8601)
-
-        authors_uids = commits.map { |commit| commit.author.id }
-        Identity.github.where(uid: authors_uids).find_each do |i|
-          BadgeAchievement
-            .where(user_id: i.user_id, badge_id: badge_id)
-            .first_or_create(rewarding_context_message_markdown: msg)
-        end
+        award_contributors(repo)
```

Comment:
After comparing award single vs award multi a bit, I realize that they are both rewarding the base `dev-contributor` which is redundant work. award multi is enough to do all the work, unless I missed something @vishaldeepak ?

Reply:
@maestromac  This should answer your question. https://github.com/forem/forem/pull/14381#discussion_r683720317

Reply:
After speaking with @djuber , I decided to undo my changes. There's limitation to the contributor API so it might be a better idea to keep it as is.